Patch-Source: https://github.com/php/php-src/commit/404803577db5e53209a5ae881c39c9092b7deacb
From 404803577db5e53209a5ae881c39c9092b7deacb Mon Sep 17 00:00:00 2001
From: Ayesh Karunaratne <ayesh@aye.sh>
Date: Thu, 1 Feb 2024 02:03:55 +0700
Subject: [PATCH] ext/curl: Fix failing tests due to string changes in libcurl
 8.6.0

Upstream libcurl 8.6.0 contains a change[^1] that caused a test failure.
This fixes it by updating the test's `EXPECTF` to use a regex to account for both string patterns.

[^1]: https://github.com/curl/curl/commit/45cf4755e71f#diff-a8a54563608f8155973318f4ddb61d7328dab512b8ff2b5cc48cc76979d4204cL1683

Closes GH-13293.
---
 NEWS                         | 2 ++
 ext/curl/tests/bug77946.phpt | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/curl/tests/bug77946.phpt b/ext/curl/tests/bug77946.phpt
index 8b1e70675554f..7b836187d6bfb 100644
--- a/ext/curl/tests/bug77946.phpt
+++ b/ext/curl/tests/bug77946.phpt
@@ -34,4 +34,4 @@ curl_multi_close($mh);
 --EXPECTF--
 int(1)
 int(1)
-string(%d) "Protocol %Sunknown%S not supported or disabled in libcurl"
+string(%d) "Protocol %Sunknown%S %rnot supported( or disabled in libcurl)?%r"
